src/test/java/org/apache/zookeeper/book/TestTaskWorkerAssignmentCallback.java

```diff
@@ -0,0 +1,38 @@
+package org.apache.zookeeper.book;
+
+import org.apache.zookeeper.KeeperException;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * A test class for mocking the Master that records the last worker for which
+ * getAbsentWorkerTasks was called.
+ */
+class TestMaster extends Master {
+    String lastWorker;
+
+    TestMaster() {
+        super("localhost:2181");
+    }
+
+    @Override
+    void getAbsentWorkerTasks(String worker) {
+        lastWorker = worker;
+    }
+}
+
+public class TestTaskWorkerAssignmentCallback {
+
+    @Test
+    public void taskWorkerAssignmentCallback() throws Exception {
+        TestMaster m = new TestMaster();
+        
+        String testWorker = "worker-001";
+        m.workerAssignmentCallback.processResult(KeeperException.Code.CONNECTIONLOSS.intValue(),
+                "/assign/" + testWorker,
+                (Object) testWorker,
```

Comment:
Just a nitpick! This casting is redundant. All Java classes implicitly extend `Object`.